lib/writer.js

```diff
@@ -147,3 +162,44 @@ function currentDate() {
 
   return util.format('%d-%s-%s', now.getFullYear(), pad(now.getMonth() + 1), pad(now.getDate()));
 }
+
+function nonGithubUrl(url) {
+  try {
+    var idx = url.indexOf("@");
+    if (idx !== -1) {
+      url = url.slice(idx + 1).replace(/:([^\d]+)/, "/$1");
+    }
+    url = url_.parse(url);
+    var protocol = url.protocol === "https:"
+      ? "https:"
+      : "http:"
+    return protocol + "//" + (url.host || "") +
+      url.path.replace(/\.git$/, "");
+  }
+  catch(e) {}
+}
+
+function getPackageData(pkg) {
+  var pkgData = JSON.parse(fs.readFileSync(pkg, {encoding:'utf8'})) || {};
+  normalizeData(pkgData);
+  return pkgData;
+}
+
+function getPackageRepository(pkgData) {
+  var repository = pkgData.repository;
+  if (!repository) {
+    return null;
+  }
+
+  var url = repository.url;
+
+  url = (url && url.indexOf("github") > -1)
+    ? parseGitHubURL(url)
+    : nonGithubUrl(url);
+
+  if (!url) {
+    return null;
+  }
+
+  return url;
```

Comment:
Or better yet, just `return url` and the falsy value will just be a falsy value.